src/util-base64.c

```diff
@@ -156,11 +156,27 @@ Base64Ecode DecodeBase64(uint8_t *dest, uint32_t dest_size, const uint8_t *src,
         }
     }
 
-    if (!valid && mode == BASE64_MODE_RFC4648) {
-        padding = B64_BLOCK - bbidx;
-        *decoded_bytes += ASCII_BLOCK - padding;
+    if (bbidx > 0 &&
+            ((!valid && mode == BASE64_MODE_RFC4648) || (valid && mode == BASE64_MODE_RFC2045))) {
+        /* Decoded bytes for 1 or 2 base64 encoded bytes is 1 */
+        padding = bbidx > 1 ? B64_BLOCK - bbidx : 2;
+        uint32_t numDecoded_blk = ASCII_BLOCK - (padding < B64_BLOCK ? padding : ASCII_BLOCK);
+        if (dest_size < *decoded_bytes + numDecoded_blk) {
+            SCLogDebug("Destination buffer full");
+            ecode = BASE64_ECODE_BUF;
+            *consumed_bytes += leading_sp;
+            return ecode;
+        }
+        /* if the destination size is not at least 3 Bytes long, it'll give a dynamic
+         * buffer overflow while decoding, so, return and let the caller take care of the
+         * remaining bytes to be decoded which should always be < 4 at this stage */
```

Comment:
this might be a nit -- the comment says "< 4" but the code has `< 3` ... should the code be `<= 3`?

Reply:
Not really. But, the comment could be clearer. 3 Bytes are the decoded bytes that need to fit in the destination buffer. < 4 Bytes comment is for the bytes to be decoded..